crates/sui-core/src/authority_server.rs

```diff
@@ -120,12 +120,18 @@ impl AuthorityServer {
     ) -> SuiResult<JoinHandle<SuiResult<()>>> {
         // Start the batching subsystem, and register the handles with the authority.
         let state = self.state.clone();
-        let _batch_join_handle =
-            tokio::task::spawn(
-                async move { state.run_batch_service(min_batch_size, max_delay).await },
-            );
+        let batch_join_handle = tokio::task::spawn(async move {
+            loop {
+                match state.run_batch_service(min_batch_size, max_delay).await {
```

Comment:
is this one actually needed? i think the one in sui-node should suffice for everything.

Reply:
Ok, i see this is only called from tests. Can you factor the duplicated code to a common location then? most active processes are managed in authority_active.rs, i think this one is weird because it predates AuthorityActive.

Reply:
Good idea, done